spring-integration-ws/src/test/java/org/springframework/integration/ws/config/UriVariableTests.java

```diff
@@ -134,7 +131,7 @@ public void testHttpUriVariables() {
 		assertThatExceptionOfType(MessagingException.class)
 				.isThrownBy(() -> this.inputHttp.send(message))
 				.withCauseInstanceOf(WebServiceIOException.class); // offline
-		assertThat(uri.get()).isEqualTo("http://localhost/spring-integration?param=test1%20&%20test2");
+		assertThat(uri.get()).isEqualTo("http://localhost/spring-integration?param=test1%20%26%20test2");
```

Comment:
Wow - looks like this was a bug! Looks like this should always have been encoded.